File: Solution/[700]Search in a Binary Search Tree.py
class Solution(object):
    def searchBST(self, root, val):
        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """
        if not root:
            return None
        elif root.val == val:
            return root
        elif root.val < val:
            return self.searchBST(root.right,val)
        else:
            return self.searchBST(root.left,val)

File: Solution/[700]Search in a Binary Search Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* searchBST(TreeNode* root, int val) {
        if(root->val == val || !root)
        {
            return root;
        }
        else if(root->val > val)
        {
            if(root->left)
            {
                return searchBST(root->left, val);
            }
        }
        else
        {
            if(root->right)
            {
                return searchBST(root->right, val);
            }
        }
        return NULL;
    }
};